tests/tests/test-revert-receipt.ts

```diff
@@ -25,9 +25,17 @@ describeWithMoonbeam("Frontier RPC (Constructor Revert)", `simple-specs.json`, (
     "80fdfea2646970667358221220c70bc8b03cdfdf57b5f6c4131b836f9c2c4df01b8202f530555333f2a00e4b8364" +
     "736f6c63430006060033";
 
+  let expected_contract_address;
+  beforeEach(async function () {
```

Comment:
It looks like this is bringing some test dependency.

Reply:
How? Before this PR the test was assuming the first test was being executed before the second (because the nonce is different and addresses are hardcoded). What I added is a check that is computed before each test that calculates the appropriate address based on the current nonce. I tested changing their order, and skipping one of the tests and it seems to be passing.

Reply:
I did this to avoid spinning up a new node just because this stupid nonce thingy. If we believe this is overkill, we can spin up a new node

Reply:
The issue is that because they are relying on a read (transaction nonce) and write (send the transaction to the address based on that nonce) and it is non-atomic (the read is asynchronous), if both test runs concurrently one might read before the other but write after, leading to incorrect state.
This is the kind of dependency we should remove (spinning up another node is fine for now, or simply using a different address so the nonce is 0 for each test)

Reply:
cool